plutus-core/testlib/UntypedPlutusCore/Test/Term/Construction.hs

```diff
@@ -0,0 +1,515 @@
+
+bls12_381_finalVerify :: T -> T -> T
+bls12_381_finalVerify ml1 ml2 =
+  builtin Bls12_381_finalVerify `app` ml1 `app` ml2
+
+mkNilPairData :: T -> T
+mkNilPairData unitVal = builtin MkNilPairData `app` unitVal
+
+keccak_256 :: T -> T
+keccak_256 = app (builtin Keccak_256)
+
+blake2b_224 :: T -> T
+blake2b_224 = app (builtin Blake2b_224)
+
+integerToByteString :: T -> T -> T -> T
+integerToByteString signed endianness i =
+  builtin IntegerToByteString `app` signed `app` endianness `app` i
+
+byteStringToInteger :: T -> T -> T
+byteStringToInteger signed bs =
+  builtin ByteStringToInteger `app` signed `app` bs
+
+andByteString :: T -> T -> T -> T
+andByteString signed bs1 bs2 =
+  builtin AndByteString `app` signed `app` bs1 `app` bs2
+
+orByteString :: T -> T -> T -> T
+orByteString signed bs1 bs2 =
+  builtin OrByteString `app` signed `app` bs1 `app` bs2
+
+xorByteString :: T -> T -> T -> T
+xorByteString signed bs1 bs2 =
+  builtin XorByteString `app` signed `app` bs1 `app` bs2
+
+complementByteString :: T -> T
+complementByteString bs =
+  builtin ComplementByteString `app` bs
+
+readBit :: T -> T -> T
+readBit bs i =
+  builtin ReadBit `app` bs `app` i
+
+writeBits :: T -> T -> T -> T
+writeBits bs indices bit =
+  builtin WriteBits `app` bs `app` indices `app` bit
+
+replicateByte :: T -> T -> T
+replicateByte count byte =
+  builtin ReplicateByte `app` count `app` byte
+
+shiftByteString :: T -> T -> T
+shiftByteString bs shift =
+  builtin ShiftByteString `app` bs `app` shift
+
+rotateByteString :: T -> T -> T
+rotateByteString bs rotate =
+  builtin RotateByteString `app` bs `app` rotate
+
+countSetBits :: T -> T
+countSetBits bs =
+  builtin CountSetBits `app` bs
+
+findFirstSetBit :: T -> T
+findFirstSetBit bs =
+  builtin FindFirstSetBit `app` bs
+
+ripemd_160 :: T -> T
+ripemd_160 bs =
+  builtin Ripemd_160 `app` bs
+
+expModInteger :: T -> T -> T -> T
+expModInteger base ex modulus =
+  builtin ExpModInteger `app` base `app` ex `app` modulus
+
+caseList :: T -> T -> T -> T
+caseList xs nilCase consCase =
+  force (builtin CaseList) `app` xs `app` nilCase `app` consCase
+
+caseData :: T -> T -> T -> T -> T -> T -> T
+caseData d constrCase mapCase listCase iCase bCase =
+  force (builtin CaseData)
+    `app` d
+    `app` constrCase
+    `app` mapCase
+    `app` listCase
+    `app` iCase
+    `app` bCase
+
+dropList :: T -> T -> T
+dropList n xs = force (builtin DropList) `app` n `app` xs
+
+lengthOfArray :: T -> T
+lengthOfArray xs = force (builtin LengthOfArray) `app` xs
+
+listToArray :: T -> T
+listToArray xs = force (builtin ListToArray) `app` xs
+
+indexArray :: T -> T -> T
+indexArray xs i = force (builtin IndexArray) `app` xs `app` i
+
+--------------------------------------------------------------------------------
+-- Unique names
+--------------------------------------------------------------------------------
+
+uniqueNames2
+  :: String
+  -> String
+  -> (Name, Name)
+uniqueNames2 n1 n2 =
+  (name n1 0, name n2 1)
```

Comment:
Second, this is a strange way of getting unique names. In most if not every other API (e.g., template haskell API, GHC API), getting uinque names is a monadic action. This way the names are truly unique. In your method, calling `uniqueNames2 "a" "b"` the second time returns the same names, which are no longer unique.